packages/next/lib/verifyTypeScriptSetup.ts

```diff
@@ -28,7 +30,9 @@ export async function verifyTypeScriptSetup(
   imageImportsEnabled: boolean,
   cacheDir?: string
 ): Promise<{ result?: TypeCheckResult; version: string | null }> {
-  const tsConfigPath = path.join(dir, 'tsconfig.json')
+  const config = await loadConfig(PHASE_PRODUCTION_BUILD, dir)
+  const tsConfigName = config.typescript.tsconfigPath
```

Comment:
Same here, the next config object should be passed in as a parameter instead of being read from the filesystem again